firewood/src/merkle.rs

```diff
@@ -51,27 +53,27 @@ fn nibbles_formatter<X: IntoIterator<Item = u8>>(nib: X) -> String {
 macro_rules! write_attributes {
     ($writer:ident, $node:expr, $value:expr) => {
         if !$node.partial_path.0.is_empty() {
-            write!(
-                $writer,
-                " pp={}",
-                nibbles_formatter($node.partial_path.0.clone())
-            )
-            .map_err(|e| FileIoError::from_generic_no_file(e, "write attributes"))?;
+            let pp_str = format!(" pp={}", nibbles_formatter($node.partial_path.0.clone()));
+            $writer
+                .write_all(pp_str.as_bytes())
```

Comment:
and then the temporary `pp_str` it's copied to the writer and then immediately dropped.

we could completely obviate the need for two heap buffers by writing directly to the writer:

```
write!($writer, " pp={:x}", $node.partial_path)
.map_err(|e| FileIoError::from_generic_no_file(e, "write attributes"))?;
```

Reply:
Moved to 1163

Reply:
Unmoved, and fixed here.